File: AAS.BackOffice/client_app/src/domain/users/userRegistrationBlank.ts
import { User } from "./user"

export interface UserRegistrationBlank {
    id: string | null,
    firstName: string | null,
    middleName: string | null,
    lastName: string | null,
    email: string | null,
    password: string | null,
    rePassword: string | null,
    phoneNumber: string | null
}

export namespace UserRegistrationBlank {
    export function getDefault(): UserRegistrationBlank {
        return {
            id: null,
            firstName: null,
            middleName: null,
            lastName: null,
            email: null,
            password: null,
            rePassword: null,
            phoneNumber: null
        }
    }
}

File: AAS.Domain/Users/UserRegistrationBlank.cs
using AAS.Tools.Managers;
using AAS.Tools.Types.IDs;

namespace AAS.Domain.Users;

public class UserRegistrationBlank
{
    public ID? Id { get; set; }
    public String? FirstName { get; set; }
    public String? MiddleName { get; set; }
    public String? LastName { get; set; }
    public String? Email { get; set; }
    public String? PhoneNumber { get; set; }
    public String? Password { get; set; }
    public String? RePassword { get; set; }

    public String? PasswordHash => HashManager.DefinePasswordHash(Password);
}